src/components/views/messages/MessageActionBar.tsx

```diff
@@ -579,6 +582,25 @@ export default class MessageActionBar extends React.PureComponent<IMessageAction
             );
         }
 
+        const highlightInfo = getEventHighlightInfo(
+            this.props.mxEvent,
+            MatrixClientPeg.get(),
+            this.context.timelineRenderingType,
+        );
+
+        if (highlightInfo.isHighlighted && highlightInfo.rule) {
+            const reason = getHighlightReasonMessage(highlightInfo.rule);
+            toolbarOpts.splice(
+                0,
+                0,
+                <TooltipTarget key="highlight-annotation" data-testid="highlight-annotation" label={reason}>
+                    <div className="mx_MessageActionBar_infoElement">
+                        <NotificationIcon />
+                    </div>
+                </TooltipTarget>,
+            );
```

Comment:
While not necessarily intuitive, you might want to use `Array.unshift` here.

| -         | Add     | Remove |
| --------- | ------- | ------ |
| **Start** | unshift | shift  |
| **End**   | push    | pop    |